core/cmd/shell_local.go

```diff
@@ -322,59 +320,50 @@ func (s *Shell) runNode(c *cli.Context) error {
 		return fmt.Errorf("failed to create root directory %q: %w", s.Config.RootDir(), err)
 	}
 
-	cfg := s.Config
-	ldb := pg.NewLockedDB(cfg.AppID(), cfg.Database(), cfg.Database().Lock(), lggr)
-
 	// rootCtx will be cancelled when SIGINT|SIGTERM is received
 	rootCtx, cancelRootCtx := context.WithCancel(context.Background())
 
 	// cleanExit is used to skip "fail fast" routine
 	cleanExit := make(chan struct{})
-	var shutdownStartTime time.Time
 	defer func() {
+		cancelRootCtx() // Ensure context is always cancelled
```

Comment:
This is the only call, so isn't the graceful timer blocked until this point, when graceful shutdown has already happened?